mqtt_client.c

```diff
@@ -990,6 +992,30 @@ static esp_err_t deliver_publish(esp_mqtt_client_handle_t client)
     return ESP_OK;
 }
 
+static esp_err_t deliver_suback(esp_mqtt_client_handle_t client)
+{
+    uint8_t *msg_buf = client->mqtt_state.in_buffer;
+    size_t msg_read_len = client->mqtt_state.in_buffer_read_len;
+    size_t msg_data_len = msg_read_len;
+    char *msg_data = NULL;
+    ESP_LOGD(TAG, "Get data len= %zu, read len= %zu", msg_data_len, msg_read_len);
+
+    msg_data = mqtt_get_suback_data(msg_buf, &msg_data_len);
+    if (msg_data_len <= 0) {
+        ESP_LOGE(TAG, "Read error or timeout: len_read=%zu, errno=%d", msg_data_len, errno);
```

Comment:
This is not a read error, no need to check for `errno`. Perhaps a more general error message, something like a _unexpected_ error :
```suggestion
        ESP_LOGE(TAG, "Failed to acquire suback data");
```